File: tests/fixtures/codegen/inputs/standalone-expected/typescript/runtime/queries/items/UpdatePayload.ts
import type { UpdatePayloadParams } from "../../../types/queries/items/UpdatePayload.js";
export type { UpdatePayloadParams } from "../../../types/queries/items/UpdatePayload.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const updatePayloadSql = "UPDATE items SET payload = $1 WHERE id = $2;";
export function isUpdatePayloadParamsPayloadPublicItemsPayload(value: unknown): value is UpdatePayloadParams["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateUpdatePayloadParamsPayloadPublicItemsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export async function updatePayload(db: pgsid.Queryable, params: UpdatePayloadParams): Promise<number> {
    const jsonInput1 = JSON.stringify(params["payload"]);
    if (jsonInput1 === undefined)
        throw new pgsid.QueryValidationError("UpdatePayload", "payload", [{ path: [], keyword: "json", expected: "JSON value", received: "undefined", message: "Value has no JSON representation" }]);
    const jsonInput1Value = jsonInput1 === null ? null : JSON.parse(jsonInput1);
    if (jsonInput1 !== null) {
        if (!isUpdatePayloadParamsPayloadPublicItemsPayload(jsonInput1Value))
            throw new pgsid.QueryValidationError("UpdatePayload", "payload", validateUpdatePayloadParamsPayloadPublicItemsPayload(jsonInput1Value).issues);
    }
    const result = await db.query(updatePayloadSql, [jsonInput1, params["id"]]);
    return result.rowCount ?? 0;
}

File: tests/fixtures/codegen/inputs/structs-expected/queries/items/updatepayload.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
	pgsid "example.com/pgsid-validation/generated/pgsid"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
)

const UpdatePayloadSQL = "UPDATE items SET payload = $1 WHERE id = $2;"

type UpdatePayloadParams struct {
	Payload jsonschemas.Event `db:"payload"`
	Id      pgsid.Null[int32] `db:"id"`
}

func (q *Queries) UpdatePayload(ctx context.Context, params UpdatePayloadParams) (int64, error) {
	jsonInput1, err := pgsidpgx.ValidateJSONInput(params.Payload, "pgsid:///jsonschemas/Event.json#", "UpdatePayload", "payload", false)
	if err != nil {
		return 0, err
	}
	tag, err := q.db.Exec(ctx, UpdatePayloadSQL, jsonInput1, pgsidpgx.Value(params.Id))
	if err != nil {
		return 0, err
	}
	return tag.RowsAffected(), nil
}
